db/db_impl.cc

```diff
@@ -674,17 +683,34 @@ void DBImpl::PersistStats() {
     return;
   }
   size_t stats_history_size_limit = 0;
+  bool persist_stats_to_disk = false;
   {
     InstrumentedMutexLock l(&mutex_);
     stats_history_size_limit = mutable_db_options_.stats_history_buffer_size;
+    persist_stats_to_disk = mutable_db_options_.persist_stats_to_disk;
   }
 
-  // TODO(Zhongyi): also persist immutable_db_options_.statistics
-  {
-    std::map<std::string, uint64_t> stats_map;
-    if (!statistics->getTickerMap(&stats_map)) {
-      return;
-    }
+  std::map<std::string, uint64_t> stats_map;
+  if (!statistics->getTickerMap(&stats_map)) {
+    return;
+  }
+  if (persist_stats_to_disk) {
+    std::string now_micros_string = rocksdb::ToString(now_micros);
+    // make time stamp string equal in length to allow sorting by time
+    now_micros_string.insert(
+        0, kTimestampStringMinimumLength - now_micros_string.length(), '0');
+    int keycount = 0;
+    WriteOptions wo;
+    // for (auto iter = stats_map.begin(); iter != stats_map.end(); ++iter) {
+    for (const auto& stat : stats_map) {
+      std::string key(now_micros_string);
+      key.append("#").append(stat.first);
```

Comment:
In this code base, we tend to use sprintf to do this kind of formatting. In this place, I also feel that it is clearer to read.

Reply:
the issue here is that key's length is unknown so we need to guess the length when defining the char array, which might waste some memory, and that's for every stat that we collect.

Reply:
It's just waste temporary memory here, not every stat. Do I understand correctly?

Reply:
yes, just temporary memory waste, will replace std::string with char array and use snprintf